Runtime/Matchers/IComponentMatcher.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2023 Koji Hasegawa.
+// This software is released under the MIT License.
+
+using UnityEngine;
+
+namespace TestHelper.Monkey.Matchers
+{
+    public interface IComponentMatcher
+    {
+        public bool IsMatch(MonoBehaviour component);
+    }
```

Comment:
The concept seems good, but I feel better that replacing an interface that have only single method with just functions if you have no plan to add methods anymore.

Reply:
Right. However, default matcher functions are written in MonkeyConfig.
I feel like there are too many volumes.
I'll think about it some more.

Reply:
How about providing named function?

```csharp
public static class ComponentMatcher
{
    public static Func<GameObject, bool> ByName(string name)
    {
        return gameObject => gameObject.name == name;
    }

    public static Func<GameObject, bool> ByAncestor(Func<GameObject, bool> condition)
    {
        return gameObject => 
        {
            var transform = gameObject.transform;
            while (transform is not null)
            {
                if (condition(transform.gameObject)) return true;
                transform = transform.parent;
            }
            return false;
        };
    }

    // ... and so on

    // also provising matcher combinators

    public static Func<GameObject, bool> ByComponent<T>()
    {
        return gameObject => gameObject.TryGetComponent<T>(out _);
    }

    public static Func<GameObject, bool> And(params Func<GameObject, bool>[] fs)
    {
        return gameObject => fs.All(f => f(gameObject));
    }

    public static Func<GameObject, bool> Or(params Func<GameObject, bool>[] fs)
    {
        return gameObject => fs.Any(f => f(gameObject));
    }
}
```

Now, we can implement custom matchers by short codes like following:

```csharp
public static class MyComponentMatcher
{
    public Func<GameObject, bool> ByButtonInDialog => ComponentMatcher.And(
        ComponentMatcher.ByComponent<MyButton>(),
        ComponentMatcher.ByAncestor(ComponentMatcher.ByName("dialog")),
    );
}
```

Reply:
Looks good! Thank you!